### ISSUE TITLE
Indent size not preserved

### ISSUE THREAD

**Issue by zgoda** [sindresorhus/atom-editorconfig#101]
I have this .editorconfig file in top-level directory:

```
root = true

[*]
charset = utf-8
end_of_line = lf
insert_final_newline = true
indent_style = space
indent_size = 4

[*.{json,yml,html,js}]
indent_size = 2

[{Makefile,*.java,*.sh}]
indent_style = tab

[*.{c,cpp,h,hpp}]
indent_style = tab
indent_size = 2
```

Now all files in subdirecotries that fall into last rule have `indent_size` of 4 instead of 2, yet `indent_style` is properly set to tab. There are no `.editorconfig` files in these subdirectories and Preferences->Editor->Tab Type is set to "auto".

Installed packages:

```
$ apm list
Built-in Atom packages (90)
├── about@1.1.0
├── archive-view@0.60.0
├── atom-dark-syntax@0.27.0
├── atom-dark-ui@0.51.0
├── atom-light-syntax@0.28.0
├── atom-light-ui@0.43.0
├── autocomplete-atom-api@0.9.2
├── autocomplete-css@0.10.1
├── autocomplete-html@0.7.2
├── autocomplete-plus@2.19.0
├── autocomplete-snippets@1.7.1
├── autoflow@0.25.0
├── autosave@0.22.0
├── background-tips@0.26.0
├── base16-tomorrow-dark-theme@0.27.0
├── base16-tomorrow-light-theme@0.9.0
├── bookmarks@0.36.0
├── bracket-matcher@0.76.0
├── command-palette@0.36.0
├── deprecation-cop@0.54.0
├── dev-live-reload@0.46.0
├── encoding-selector@0.21.0
├── exception-reporting@0.36.0
├── find-and-replace@0.180.0
├── fuzzy-finder@0.88.0
├── git-diff@0.55.0
├── go-to-line@0.30.0
├── grammar-selector@0.47.0
├── image-view@0.54.0
├── incompatible-packages@0.24.1
├── keybinding-resolver@0.33.0
├── language-c@0.47.1
├── language-clojure@0.16.0
├── language-coffee-script@0.41.0
├── language-csharp@0.7.0
├── language-css@0.33.0
├── language-gfm@0.81.0
├── language-git@0.10.0
├── language-go@0.37.0
├── language-html@0.41.2
├── language-hyperlink@0.14.0
├── language-java@0.16.0
├── language-javascript@0.92.0
├── language-json@0.16.0
├── language-less@0.28.2
├── language-make@0.17.0
├── language-mustache@0.12.0
├── language-objective-c@0.15.0
├── language-perl@0.28.0
├── language-php@0.29.0
├── language-property-list@0.8.0
├── language-python@0.39.0
├── language-ruby@0.57.0
├── language-ruby-on-rails@0.22.0
├── language-sass@0.40.1
├── language-shellscript@0.15.0
├── language-source@0.9.0
├── language-sql@0.17.0
├── language-text@0.7.0
├── language-todo@0.27.0
├── language-toml@0.16.0
├── language-xml@0.32.0
├── language-yaml@0.24.0
├── line-ending-selector@0.0.5
├── link@0.30.0
├── markdown-preview@0.150.0
├── metrics@0.51.0
├── notifications@0.59.0
├── one-dark-syntax@1.1.0
├── one-dark-ui@1.0.3
├── one-light-syntax@1.1.0
├── one-light-ui@1.0.3
├── open-on-github@0.38.0
├── package-generator@0.40.0
├── release-notes@0.53.0
├── settings-view@0.216.0
├── snippets@0.95.0
├── solarized-dark-syntax@0.38.1
├── solarized-light-syntax@0.22.1
├── spell-check@0.59.0
├── status-bar@0.79.0
├── styleguide@0.44.0
├── symbols-view@0.104.0
├── tabs@0.84.0
├── timecop@0.31.0
├── tree-view@0.186.0
├── update-package-dependencies@0.10.0
├── welcome@0.30.0
├── whitespace@0.31.0
└── wrap-guide@0.36.0

/home/jazg/.atom/packages (18)
├── autocomplete-clang@0.10.0
├── base16-syntax@1.7.0
├── build@0.65.0
├── busy@0.7.0
├── editorconfig@1.4.1
├── kary-foundation-dark@4.0.0
├── language-ini@1.16.0
├── language-restructuredtext@1.0.1
├── linter@1.11.18
├── linter-flake8@2.0.0
├── linter-gcc@0.6.15
├── linter-python-flake8@1.0.7
├── minimap@4.25.0
├── monokai@0.18.0
├── platformio-ide@1.6.0
├── platformio-ide-terminal@2.2.0
├── railscast-theme@0.3.0
└── tool-bar@1.0.1
```

**Comment by florianb** [comment]
Hi @zgoda, thank you very much for reporting! :)

Since your problem affects subdirectories i want to suggesr trying to address

```
[**.{c,cpp,h,hpp}]
```

Since the single asterisk matches" any string of characters, except path separators (/)", this should be the reason why your subdirectories aren't affected. However, the editorconfig-parsing is done by https://github.com/editorconfig/editorconfig-core-js, so if you identified a bug it will likely belong into that repository.

If this issue isn't resolved for you please come back, it then would be helpful to have a full file-tree of the given project (`tree -a .`). 🍻

**Comment by zgoda** [comment]
Thanks, I am almost sure it worked until last week or so. ;)

**Comment by florianb** [comment]
@zgoda -- you are welcome.. 😄  You would save me some time if you verify it.

**Comment by zgoda** [comment]
Thanks. One thing seems at least strange that `indent_style` for eg. *.cpp files is set properly yet `indent_size` is not. The parser did not change in last months, your plugin code for weeks and the only thing that changed in my instance is Atom update from 1.11.0 to 1.11.1 - changelog shows this update was [related to intendation bug](https://github.com/atom/atom/issues/12575) and this may be some hint.

**Comment by florianb** [comment]
Hmm.. interesting.. I don't see that this change is affecting the work of atom-editorconfig, since we're resetting the tabType based on the configuration, which then should be recognized properly.

It takes me a bit of time to reread the whole implementation of the tabs to see if there were any other changes introduced.

I would like to ask you to set the tab-type of the last setting to spaces and take a look if the rule gets applied  in general (or if the tab-setting comes from elsewhere).

**Comment by zgoda** [comment]
I ran Atom in safe mode for a moment to see if I perhaps messed whitespace in my files. After that right intendation was back with the same `.editorconfig` I had. I guess this issue was somehow related to Atom update (perhaps something was reset in configuration?) and the setting for open file was preserved. Now that I reopened all files right settings parsed from `.editorconfig` have been set.

I guess it is safe to close this issue, thank you for your support.
